src/Datasource/Paging/PaginatorInterface.php

```diff
@@ -41,3 +41,8 @@ public function paginate(object $object, array $params = [], array $settings = [
      */
     public function getPagingParams(): array;
 }
+
+// phpcs:disable
+// The old interface Cake\Datasource\PaginatorInterface will not load during instanceof / type checks so ensure it's loaded here
+class_alias('Cake\Datasource\Paging\PaginatorInterface', 'Cake\Datasource\PaginatorInterface');
```

Comment:
This looks good. We should change PagintorInterface to call class_exists() since the aliasing will fail if both files are loaded. This is how we solve this in other places.